src/client.ts

```diff
@@ -117,6 +125,18 @@ export default class Client implements ClientInterface {
         formatter: configuration.get("formatter"),
       },
       middleware: {
+        provideCodeLenses: async (document, token, next) => {
+          if (this.client) {
+            const response = await next(document, token);
+
+            if (response) {
+              this.testController.createTestItems(response as CodeLens[]);
+            }
+
+            return response;
+          }
+          return null;
```

Comment:
Nitpick, can we invert the check to reduce nesting here?
```suggestion
          if (!this.client) {
            return null;
          }
          
          const response = await next(document, token);

          if (response) {
            this.testController.createTestItems(response as CodeLens[]);
          }

          return response;
```